backend/src/worker_flow.rs

```diff
@@ -865,14 +861,52 @@ async fn push_next_flow_job(
     }
 }
 
+async fn get_transform_context(
+    db: &DB,
+    flow_job: &QueuedJob,
+    status: &FlowStatus,
+    transform_context: &mut Option<(String, Vec<String>)>,
+) -> error::Result<(String, Vec<String>)> {
+    if transform_context.is_none() {
```

Comment:
I suspect it's possible to remove the `transform_context` argument, remove this if condition, and call the function by writing `transform_context = transform_context.unwrap_or_else(|| get_transform_context(...))`.  That might be a bit nicer.

Reply:
lol you're absolutely right, I went too far in the "if only we had lazy val like in scala" path

Reply:
Closures, like in `unwrap_or_else` are a common way to avoid doing computation up-front. The `unwrap_or` function takes a value, `unwrap_or_else` computes the value from a closure. So maybe it's doing what you want in terms of laziness. (But the way it is here always `clone()`s the `transform_context` when returning although this is probably optimized away etc....)